lib/ParseSIL/ParseSIL.cpp

```diff
@@ -885,6 +887,74 @@ void SILParser::convertRequirements(SILFunction *F,
   }
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+/// Parse a SIL constant value (one of the following categories):
+/// - An integer literal (42).
+/// - A floating point literal (3.14).
+/// - A metatype (the instance type is parsed) ($Float).
+/// - An aggregate ([1, 2, 3]).
+/// Returns true on error.
+static bool parseSymbolicValue(SymbolicValue &value, SILParser &SP,
+                               SILBuilder &B) {
+  auto &P = SP.P;
+  auto &allocator = B.getModule().getASTContext().getAllocator();
+  // Handle integer literals.
+  if (P.Tok.is(tok::integer_literal)) {
+    APInt intValue;
+    if (SP.parseInteger(intValue, diag::sil_expected_graph_op_attr_value))
+      return true;
+    value = SymbolicValue::getInteger(intValue, allocator);
+    return false;
+  }
+  // Handle floating point literals.
+  if (P.Tok.is(tok::floating_literal)) {
+    double floatValue;
+    P.Tok.getText().getAsDouble(floatValue);
+    P.consumeToken(tok::floating_literal);
+    value = SymbolicValue::getFloat(APFloat(floatValue), allocator);
+    return false;
+  }
+  // Handle string literalss.
+  if (P.Tok.is(tok::string_literal)) {
+    // TODO: Uncomment when `getStringValue` is implemented.
+    // StringRef rawString = P.Tok.getText().drop_front().drop_back();
+    // value = SymbolicValue::getStringValue(rawString, allocator);
+    // return false;
+    llvm_unreachable("`SymbolicValue.getStringValue` is unimplemented");
```

Comment:
Rather than making it trap I would just emit a diagnostic.

Reply:
Done.